proxy/src/main/java/com/velocitypowered/proxy/server/VelocityRegisteredServer.java

```diff
@@ -73,9 +75,30 @@ public class VelocityRegisteredServer implements RegisteredServer, ForwardingAud
   private final ServerInfo serverInfo;
   private final Map<UUID, ConnectedPlayer> players = new ConcurrentHashMap<>();
 
+  /**
+   * Register a backend server.
+   *
+   * @param server velocity instance
+   *
+   * @param serverInfo info of the backend server
+   */
   public VelocityRegisteredServer(@Nullable VelocityServer server, ServerInfo serverInfo) {
     this.server = server;
     this.serverInfo = Preconditions.checkNotNull(serverInfo, "serverInfo");
+    validateForwardingRequisites();
+  }
+
+  private void validateForwardingRequisites() {
+    ServerInfoForwardingMode mode = serverInfo.getServerInfoForwardingMode();
+    if (mode != ServerInfoForwardingMode.MODERN
+            && mode != ServerInfoForwardingMode.BUNGEEGUARD) {
+      return;
+    }
+    VelocityConfiguration configuration = server.getConfiguration();
+    if (configuration.getForwardingSecret() == null || configuration.getForwardingSecret().length == 0) {
+      throw new IllegalStateException("You don't have a forwarding secret set. This is required if "
+              + "you are using MODERN or BUNGEEGUARD forwarding modes.");
+    }
```

Comment:
I meant that this method should be transferred to the [`VelocityConfiguration#validate()`](https://github.com/PaperMC/Velocity/blob/dev/3.0.0/proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java#L138) method, so that no value is applied if the configuration is reloaded and validated as invalid.
Putting it here will cause any data configured before verifying that the entire server configuration section is valid to be applied